pkg/virt-api/api.go

```diff
@@ -112,18 +308,277 @@ func addInfoToSwaggerObject(swo *openapispec.Swagger) {
 	}
 }
 
-func (app *VirtAPIApp) Run() {
-	log.Fatal(http.ListenAndServe(app.Address(), nil))
+func (app *virtAPIApp) getClientCert() error {
+	authConfigMap, err := app.virtCli.CoreV1().ConfigMaps(metav1.NamespaceSystem).Get("extension-apiserver-authentication", metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	clientCA, ok := authConfigMap.Data["client-ca-file"]
+	if !ok {
+		return fmt.Errorf("client-ca-file value not found in auth config map.")
+	}
+	app.clientCABytes = []byte(clientCA)
+
+	// request-header-ca-file doesn't always exist in all deployments.
+	// set it if the value is set though.
+	requestHeaderClientCA, ok := authConfigMap.Data["requestheader-client-ca-file"]
+	if ok {
+		app.requestHeaderClientCABytes = []byte(requestHeaderClientCA)
+	}
+	return nil
+}
+
+func getNamespace() string {
+	if data, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace"); err == nil {
```

Comment:
We have already some logic based on this in virt-controller for ha. There it falls back to kube-system, if  not set. Do we want to do the same here? Otherwise it is hard to run the controller from outside of the cluster.

Reply:
that's what this function does already. look at the return at the end of the function. If no namespace is found, it's returning kube-system.